hazelcast/listener.py

```diff
@@ -0,0 +1,163 @@
+import logging
+import threading
+from uuid import uuid4
+
+from hazelcast.exception import OperationTimeoutError, HazelcastError
+from hazelcast.util import current_time_in_millis, check_not_none
+
+
+class ListenerRegistration(object):
+    def __init__(self, registration_request, decode_register_response, encode_deregister_request, handler):
+        self.registration_request = registration_request
+        self.decode_register_response = decode_register_response
+        self.encode_deregister_request = encode_deregister_request
+        self.handler = handler
+        self.connection_registrations = {}  # Dict of Connection, EventRegistration
+
+
+class EventRegistration(object):
+    def __init__(self, server_registration_id, correlation_id):
+        self.server_registration_id = server_registration_id
+        self.correlation_id = correlation_id
+
+
+class ListenerService(object):
+    logger = logging.getLogger("HazelcastClient.ListenerService")
+
+    def __init__(self, client):
+        self._client = client
+        self._invocation_service = client.invoker
+        self.is_smart = client.config.network_config.smart_routing
```

Comment:
`is_smart ` can be made private

Reply:
Although it looks like the only usages of is_smart is under ListenerService, Proxy also accesses that field:
self._is_smart = client.listener.is_smart
That's why I left it public.